src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -202,84 +202,92 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
                BigQueryStrings.toString(tableRef), gcsPaths.size(), gcsPaths.isEmpty() ? "(empty)" : gcsPaths.get(0),
                true);
 
-      // Create load conf with minimal requirements.
-      JobConfigurationLoad loadConfig = new JobConfigurationLoad();
-      loadConfig.setSchema(schema);
-      loadConfig.setSourceFormat(sourceFormat.getFormatIdentifier());
-      loadConfig.setSourceUris(gcsPaths);
-      loadConfig.setWriteDisposition(writeDisposition);
-      loadConfig.setUseAvroLogicalTypes(true);
-      if (!tableExists && createPartitionedTable) {
-        TimePartitioning timePartitioning = new TimePartitioning();
-        timePartitioning.setType("DAY");
-        if (partitionByField != null) {
-          timePartitioning.setField(partitionByField);
-        }
-        timePartitioning.setRequirePartitionFilter(requirePartitionFilter);
-        loadConfig.setTimePartitioning(timePartitioning);
-        if (!clusteringOrderList.isEmpty()) {
-          Clustering clustering = new Clustering();
-          clustering.setFields(clusteringOrderList);
-          loadConfig.setClustering(clustering);
+      if (!gcsPaths.isEmpty()) {
```

Comment:
it's hard to read if there is a bunch of code in an if block. Instead, do checks in their own if-block, and leave normal logic outside.

```
if (gcsPaths.isEmpty()) {
   // handle scenario
}

// normal logic
```

Reply:
Done.